custom/icds_reports/utils.py

```diff
@@ -439,16 +439,22 @@ def zip_folder(pdf_files):
         zip_file.writestr('ISSNIP_monthly_register_{}.pdf'.format(pdf_file['location_name']), file)
     zip_file.close()
     client.set(zip_hash, in_memory.getvalue())
+    client.expire(zip_hash, 24 * 60 * 60)
     return zip_hash
 
 
 def create_pdf_file(pdf_hash, pdf_context):
     template = get_template("icds_reports/icds_app/pdf/issnip_monthly_register.html")
     resultFile = cStringIO()
     client = get_redis_client()
+    try:
+        pdf_page = template.render(pdf_context)
+    except Exception as ex:
```

Comment:
Are you pretty confident this is where the exception occurs? I generally think of django template rendering as pretty fault tolerant.  My guess would be that errors would occur either during the call to `CreatePDF` or when assembling the `pdf_context`.  You can also use `notify_exception` to assure that we get a traceback with relevant info, although I'm not sure if you have direct access to that.

Reply:
@esoergel I am 100% sure that the problem is in evaluating ```pdf_context``, when we try to access to specific fields. From functionality context, we do this only in render function that's why I added try/except block in this lines.

Reply:
Makes sense, thanks for clarifying!